src/app/services/sms/sms.errors.ts

```diff
@@ -8,3 +8,9 @@ export class SmsSendError extends ApplicationError {
     super(message, meta)
   }
 }
+
+export class InvalidNumberError extends ApplicationError {
```

Comment:
if `sendVerificationOtp` encounters this error now, what status will be returned? I noticed `handleError` in `verification.controller.ts` doesn't have a case for `VfnErrors.InvalidMobileNumber`, so does that mean we will return 500 when we don't intend to?

Reply:
the returned error gets mapped in `verification.service#198` which converts the error returned to a thrown one that hopefully gets resolved in your vfn refactor PR.

```ts
.mapErr((err) => {
        throwError(err.message, VfnErrors.SendOtpFailed)
      })
```

Reply:
should i change the error thrown according to the error returned? i.e.

```ts
.mapErr((err) => {
  if (err instanceof InvalidNumberError) {
        throwError(err.message, VfnErrors.InvalidOtp)
  } else {
        throwError(err.message, VfnErrors.SendOtpFailed)
  }
})
```

Reply:
Or I can wait for your PR to then fix this, since SendOtpFailed is the old behaviour:
```ts
try {
      await sendOTPForField(formId, field, answer, otp)
    } catch (err) {
      throwError(err.message, VfnErrors.SendOtpFailed)
    }
```

Reply:
nvm haha I think it's best to resolve it in my PR